lib/SetAPI/readsalignment/ReadsAlignmentSetInterfaceV1.py

```diff
@@ -2,162 +2,257 @@
             {"objects": ref_list, "includeMetadata": 1}
         )
         num_genomes = len(set([item[10]["genome_id"] for item in info["infos"]]))
-        if num_genomes == 0 or num_genomes > 1:
-            raise ValueError(
-                "All ReadsAlignments in the set must be aligned "
-                "against the same genome reference."
-            )
+        if num_genomes != 1:
+            raise ValueError(same_ref(self.set_items_type()))
+
+    def get_reads_alignment_set(
+        self: "ReadsAlignmentSetInterfaceV1", _, params: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Retrieve reads alignment sets.
 
-    def get_reads_alignment_set(self, ctx, params):
-        """
         If the set is a KBaseSets.ReadsAlignmentSet, it gets returned as-is.
         If it's a KBaseRNASeq.RNASeqAlignmentSet, a few things get juggled.
         1. We try to figure out the object references for the alignments (which are optional)
         2. From each ref, we try to figure out the condition, and apply those as labels (also
            might be optional)
+
+        :param self: this class
+        :type self: ReadsAlignmentSetInterfaceV1
+        :param _: unused (KBase context)
+        :type ctx: dict[str, Any]
+        :param params: dictionary of parameters
+        :type params: dict[str, Any]
+        :return: results of the get_set query
+        :rtype: dict[str, Any]
         """
-        set_type, obj_spec = self._check_get_reads_alignment_set_params(params)
+        checked_params = self._check_get_set_params(params)
 
         include_item_info = False
-        if "include_item_info" in params:
-            if params["include_item_info"] == 1:
-                include_item_info = True
+        if "include_item_info" in params and params["include_item_info"] == 1:
+            include_item_info = True
 
         include_set_item_ref_paths = False
-        if "include_set_item_ref_paths" in params:
-            if params["include_set_item_ref_paths"] == 1:
-                include_set_item_ref_paths = True
+        if (
+            "include_set_item_ref_paths" in params
+            and params["include_set_item_ref_paths"] == 1
+        ):
+            include_set_item_ref_paths = True
 
         ref_path_to_set = []
         if "ref_path_to_set" in params and len(params["ref_path_to_set"]) > 0:
             ref_path_to_set = params["ref_path_to_set"]
 
+        assert checked_params[REF_PATH_TO_SET] == ref_path_to_set
+        assert checked_params[INC_ITEM_INFO] == include_item_info
+        assert checked_params[INC_ITEM_REF_PATHS] == include_set_item_ref_paths
+        assert checked_params["ref"] == params["ref"]
+
+        # check the type of the set itself
+        obj_ref_dict = build_ws_obj_selector(
+            checked_params["ref"], checked_params[REF_PATH_TO_SET]
+        )
+
+        info = self.workspace_client.get_object_info3({"objects": [obj_ref_dict]})
+        set_type = info["infos"][0][2]
```

Comment:
this is extra code that doesn't appear in the AssemblySetInterface as this class can fetch two types of set